game/plugin/navigation/door/src/door.kt

```diff
@@ -0,0 +1,180 @@
+                Direction.WEST to Direction.SOUTH,
+                Direction.EAST to Direction.NORTH
+        )
+
+        val RIGHT_HINGE_ORIENTATION: HashMap<Direction, Direction> = hashMapOf(
+                Direction.NORTH to Direction.EAST,
+                Direction.SOUTH to Direction.WEST,
+                Direction.WEST to Direction.NORTH,
+                Direction.EAST to Direction.SOUTH
+        )
+
+        val toggledDoors: HashMap<GameObject, GameObject> = hashMapOf()
+
+        val LEFT_HINGED = setOf(1516, 1536, 1533)
+
+        val RIGHT_HINGED = setOf(1519, 1530, 4465, 4467, 3014, 3017, 3018, 3019)
+
+        /**
+         * Find a given door in the world
+         * @param world The [World] the door lives in
+         * @param position The [Position] of the door
+         * @param objectId The [GameObject] id of the door
+         */
+        fun find(world: World, position: Position, objectId: Int): Door? {
+            val region = world.regionRepository.fromPosition(position)
+            val gameObject = region.findObject(position, objectId).orElseGet(null)
+            return if (gameObject == null) {
+                null
+            } else {
+                Door(gameObject)
+            }
+        }
+    }
+
+    /**
+     * Returns the supported doors by the system
+     * See [DoorType]
+     */
+    fun supported(): Boolean {
+        return type() !== DoorType.NOT_SUPPORTED
+    }
+
+
+    /**
+     * Computes the given door type by which id exists in
+     * the supported left and right hinged doors
+     */
+    fun type(): DoorType {
+        return when {
+            gameObject.id in LEFT_HINGED -> DoorType.LEFT
+            gameObject.id in RIGHT_HINGED -> DoorType.RIGHT
+            else -> DoorType.NOT_SUPPORTED
+        }
+    }
+
+    /**
+     * Toggles a given [GameObject] orientation and position
+     * Stores the door state in toggleDoors class variable
+     */
+    fun toggle() {
+        val world = gameObject.world
+        val regionRepository = world.regionRepository
+
+        // Remove current door from region
+        regionRepository.fromPosition(gameObject.position).removeEntity(gameObject)
+
+        // Checks if door has been toggled
+        val originalDoor: GameObject? = toggledDoors[gameObject]
+
+        if (originalDoor == null) {
+            // If not toggled it will change orientation and position of door
+            val position = movePosition()
+            val orientation: Int = translateDirection()?.toOrientationInteger() ?: gameObject.orientation
+
+            val toggledDoor = DynamicGameObject.createPublic(world, gameObject.id, position, gameObject.type, orientation)
+
+            // Renders new door
+            regionRepository.fromPosition(position).addEntity(toggledDoor)
+            toggledDoors.put(toggledDoor, gameObject)
+        } else {
+            // Removes toggled door
+            toggledDoors.remove(gameObject)
+            // Renders original door
```

Comment:
Comments should explain why a code is doing something, not describe what it is doing, so you can delete the inline ones (the `//` ones) here. If everything you get from a comment you can get just from reading the code, you don't need the comment.